### PR TITLE
SideBar with FILTERING

### PR DESCRIPTION
**Pull request by Patr0sss**

### CODE REVIEW

File: public/svg/Arrow.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      style={{ paddingRight: "0px" }}
+    >
+      <g id="SVGRepo_bgCarrier" stroke-width="0"></g>
+      <g
+        id="SVGRepo_tracerCarrier"
+        stroke-linecap="round"
+        stroke-linejoin="round"
+      ></g>
+      <g id="SVGRepo_iconCarrier">
+        {" "}
```
Comment (DjDister) [inline public/svg/Arrow.tsx]: remove empty spaces

File: src/components/SideBarComponents/SideBar.tsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
+"use client";
+import { useState } from "react";
+import styles from "./SideBar.module.css";
+import SideBarItem from "./SideBarItem";
+import SideSearch from "./SideSearch";
+
+function SideBar() {
+  const data = [
```
Comment (DjDister) [inline src/components/SideBarComponents/SideBar.tsx]: move const  to utils -> consts -> sidebar.consts.ts

File: src/components/SideBarComponents/SideBar.tsx [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  const pullInputData = (inputData: string) => {
+    setSearchData(inputData);
+    return inputData;
+  };
+
+  const filteredData = data.filter((algo) =>
+    algo.category.includes(searchData)
+  );
+
+  pullInputData;
```
Comment (DjDister) [inline src/components/SideBarComponents/SideBar.tsx]: ??

File: src/components/SideBarComponents/SideBar.tsx [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+        {
+          name: "Linear Search",
+          path: "linear-search",
+        },
+      ],
+    },
+  ];
+
+  const [searchData, setSearchData] = useState<string>("");
+  const pullInputData = (inputData: string) => {
```
Comment (DjDister) [inline src/components/SideBarComponents/SideBar.tsx]: rename functions so they have naming acording to their functionality for example: handleInputChange

File: src/components/SideBarComponents/SideBarItem.tsx [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+  elementsArray: { name: string; path: string }[];
+  prefixPath: string;
+}) {
+  const [itemOpened, setItemOpened] = useState<boolean>(false);
+  return (
+    <>
+      <div
+        className={styles.item}
+        onClick={() => {
+          setItemOpened(!itemOpened);
```
Comment (DjDister) [inline src/components/SideBarComponents/SideBarItem.tsx]: do not override state this way better use setState((prev) => !prev)

File: src/components/SideBarComponents/SideBarItem.tsx [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+function SideBarItem({
+  itemName,
+  elementsArray,
+  prefixPath,
+}: {
+  itemName: string;
+  elementsArray: { name: string; path: string }[];
+  prefixPath: string;
+}) {
+  const [itemOpened, setItemOpened] = useState<boolean>(false);
```
Comment (DjDister) [inline src/components/SideBarComponents/SideBarItem.tsx]: bool values should be isItemOpened and so on

File: src/components/SideBarComponents/SideBarItem.tsx [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+            }}
+            className={styles.arrowBox}
+          >
+            <Arrow width={10} />
+          </div>
+        </div>
+      </div>
+      {itemOpened ? (
+        <div
+          style={{ borderBottom: "0.5px solid white" }}
```
Comment (DjDister) [inline src/components/SideBarComponents/SideBarItem.tsx]: move to styles

File: src/components/SideBarComponents/SideBarItem.tsx [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+        onClick={() => {
+          setItemOpened(!itemOpened);
+        }}
+        style={{
+          borderBottom: itemOpened ? "none" : "0.5px solid white",
+          backgroundColor: itemOpened ? "#2e2e2e" : "#393939",
+        }}
+      >
+        <div
+          style={{
```
Comment (DjDister) [inline src/components/SideBarComponents/SideBarItem.tsx:30]: move tos tyles

File: src/components/SideBarComponents/SideSearch.tsx [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+import Magnifier from "../../../public/svg/Magnifier";
+import styles from "./SideSearch.module.css";
+
+function SideSearch({
+  inputData,
+}: {
+  inputData: (inputData: string) => string;
+}) {
+  return (
+    <div
```
Comment (DjDister) [inline src/components/SideBarComponents/SideSearch.tsx]: move to styles

**Review by DjDister — CHANGES_REQUESTED** [review]

File: public/svg/Arrow.tsx (hunk 1 above)
Comment (Patr0sss) [reply public/svg/Arrow.tsx]: Done

File: src/components/SideBarComponents/SideBar.tsx (hunk 2 above)
Comment (Patr0sss) [reply src/components/SideBarComponents/SideBar.tsx]: ?

File: src/components/SideBarComponents/SideBar.tsx (hunk 3 above)
Comment (Patr0sss) [reply src/components/SideBarComponents/SideBar.tsx]: Done

File: src/components/SideBarComponents/SideBar.tsx (hunk 4 above)
Comment (Patr0sss) [reply src/components/SideBarComponents/SideBar.tsx]: Done

File: src/components/SideBarComponents/SideBarItem.tsx (hunk 5 above)
Comment (Patr0sss) [reply src/components/SideBarComponents/SideBarItem.tsx]: K

File: src/components/SideBarComponents/SideBarItem.tsx (hunk 6 above)
Comment (Patr0sss) [reply src/components/SideBarComponents/SideBarItem.tsx]: well well well

File: src/components/SideBarComponents/SideBarItem.tsx (hunk 7 above)
Comment (Patr0sss) [reply src/components/SideBarComponents/SideBarItem.tsx]: ez

File: src/components/SideBarComponents/SideSearch.tsx (hunk 9 above)
Comment (Patr0sss) [reply src/components/SideBarComponents/SideSearch.tsx]: ok

File: src/components/SideBarComponents/SideBarItem.tsx (hunk 8 above)
Comment (Patr0sss) [reply src/components/SideBarComponents/SideBarItem.tsx:30]: ok

File: src/components/SideBarComponents/SideBar.tsx (hunk 2 above)
Comment (Patr0sss) [reply src/components/SideBarComponents/SideBar.tsx]: g

File: src/utils/consts/sideBar.consts.ts [hunk 10]
```diff
@@ -0,0 +1,31 @@
+export const sideBarData = [
```
Comment (DjDister) [inline src/utils/consts/sideBar.consts.ts]: const should be uppercased so SIDE_BAR_DATA

Comment (Patr0sss) [reply src/utils/consts/sideBar.consts.ts]: aha , dobrze że wcześniej o tym wspomniałeś

**Review by DjDister — APPROVED** [review]

**Pull request merged**